packages/react/src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -290,262 +243,241 @@ describe('Dropdown', () => {
+      fireEvent.keyDown(dropdownRoot, { which: KeyCodes.escape });
+      fireEvent.focus(dropdownRoot);
 
-      dropdownElement = document.querySelector('.ms-Dropdown-items') as HTMLElement;
-      expect(dropdownElement).toBeFalsy();
+      expect(() => getByRole('listbox')).toThrow();
     });
 
     it('uses item title attribute if provided', () => {
       const options: IDropdownOption[] = [{ key: 'a', text: 'a', title: 'b' }];
-      wrapper = mount(<Dropdown options={options} />);
+      const { getByRole } = render(<Dropdown options={options} />);
 
-      wrapper.find('.ms-Dropdown').simulate('click');
+      userEvent.click(getByRole('combobox'));
 
-      const item = wrapper.find('.ms-Dropdown-item');
-      expect(item.getElements()[0].props.title).toBe('b');
+      const item = getByRole('option');
+      expect(item.title).toBe('b');
     });
 
     // This is a way to effectively disable setting a title
     it('uses empty string if provided for title', () => {
       const options: IDropdownOption[] = [{ key: 'a', text: 'a', title: '' }];
-      wrapper = mount(<Dropdown options={options} />);
+      const { getByRole } = render(<Dropdown options={options} />);
 
-      wrapper.find('.ms-Dropdown').simulate('click');
+      userEvent.click(getByRole('combobox'));
 
-      const item = wrapper.find('.ms-Dropdown-item');
-      expect(item.getElements()[0].props.title).toBe('');
+      const item = getByRole('option');
+      expect(item.title).toBe('');
     });
   });
 
   describe('multi-select', () => {
     it('Renders correctly', () => {
-      component = create(<Dropdown options={DEFAULT_OPTIONS} multiSelect />);
-      const tree = component.toJSON();
-      expect(tree).toMatchSnapshot();
+      const { container } = render(<Dropdown options={DEFAULT_OPTIONS} multiSelect />);
+      expect(container).toMatchSnapshot();
     });
 
     it('Renders correctly when open', () => {
-      // Mock createPortal so that the options list ends up inside the wrapper for snapshotting
-      spyOn(ReactDOM, 'createPortal').and.callFake(node => node);
       // There's intermittent variation (maybe measurement-related) on different computers,
       // so use fake timers to make it more predictable even though we never advance the timers.
       jest.useFakeTimers();
 
-      const ref = React.createRef<IDropdown>();
       // Specify dropdownWidth to prevent inconsistent calculated widths from getting into the snapshot
-      wrapper = mount(<Dropdown multiSelect options={RENDER_OPTIONS} componentRef={ref} dropdownWidth={200} />);
-      ref.current!.focus(true);
-      wrapper.update();
-      expect(wrapper.getDOMNode()).toMatchSnapshot();
+      const { getByRole } = render(<Dropdown multiSelect options={RENDER_OPTIONS} dropdownWidth={200} />);
+      userEvent.click(getByRole('combobox'));
+      // snapshot the whole body to capture both the field and the options list
+      expect(document.body).toMatchSnapshot();
```

Comment:
Do you think it might be better to use [`baseElement`](https://testing-library.com/docs/react-testing-library/api#baseelement-1) rather than `document.body` for these cases?

Reply:
The layer for the options would still get attached to `document.body`, so unless I bring back the code mocking `createPortal` (which I'd prefer not to) that wouldn't help.